### PR TITLE
Add component specific presubmit targets

### PR DESCRIPTION
**Pull request by ozevren**
Add component specific presubmit targets.

+ Also add shells scripts for invoking the Galley presubmit targets in Prow.
+ Subsumes https://github.com/istio/istio/pull/13852

### CODE REVIEW

File: prow/integ-galley-k8s-tests.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
```
Comment (utka) [inline prow/integ-galley-k8s-tests.sh]: Can you reuse lib.sh instead of copy code?

File: prow/integ-galley-local-tests.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```
Comment (utka) [inline prow/integ-galley-local-tests.h]: Should most of this file be in separate common file and reused here and another files?

File: tests/integration/tests.mk [hunk 3]
```diff
@@ -105,22 +115,13 @@ test.integration.kube: | $(JUNIT_REPORT)
...
-	2>&1 | tee >($(JUNIT_REPORT) > $(JUNIT_UNIT_TEST_XML))
-
 # Presubmit integration tests targeting Kubernetes environment.
 .PHONY: test.integration.kube.presubmit
 test.integration.kube.presubmit: | $(JUNIT_REPORT)
 	mkdir -p $(dir $(JUNIT_UNIT_TEST_XML))
 	set -o pipefail; \
 	$(GO) test -p 1 ${T} ${TEST_PACKAGES} ${_INTEGRATION_TEST_WORKDIR_FLAG} ${_INTEGRATION_TEST_CIMODE_FLAG} -timeout 30m \
-    --istio.test.select +presubmit \
+    --istio.test.select -postsubmit,-flaky \
```
Comment (howardjohn) [inline tests/integration/tests.mk]: Nice change, should reduce test time by preventing unneeded Istio deployments

File: tests/integration/tests.mk [hunk 4]
```diff
@@ -130,18 +131,11 @@ test.integration.kube.presubmit: | $(JUNIT_REPORT)
...
-	$(GO) test -p 1 ${T} ${TEST_PACKAGES} ${_INTEGRATION_TEST_WORKDIR_FLAG} ${_INTEGRATION_TEST_CIMODE_FLAG} -timeout 30m \
-    --istio.test.select -presubmit,-postsubmit \
- 	--istio.test.env kube \
-	--istio.test.kube.config ${INTEGRATION_TEST_KUBECONFIG} \
-	--istio.test.hub=${HUB} \
-	--istio.test.tag=${TAG} \
-	--istio.test.pullpolicy=${_INTEGRATION_TEST_PULL_POLICY} \
-	${_INTEGRATION_TEST_INGRESS_FLAG} \
-	${_INTEGRATION_TEST_WORK_DIR_FLAG} \
+	$(GO) test -race -p 1 ${T} ${TEST_PACKAGES} -timeout 120m \
```
Comment (howardjohn) [inline tests/integration/tests.mk]: The tests take like 1-5min in my experience (not actually that variable, I just forgot), so we may want to bump the timeout down. Not a big deal though - there are benefits to having a large timeout as well

File: prow/integ-galley-k8s-tests.sh (hunk 1 above)
Comment (ozevren) [reply prow/integ-galley-k8s-tests.sh]: I don't think any of the code here is in lib.sh?

File: prow/integ-galley-local-tests.h (hunk 2 above)
Comment (ozevren) [reply prow/integ-galley-local-tests.h]: Created a common set of files and called specific targets from there (ala e2e shell file).

**Comment by ozevren** [comment]
> Are we just adding a galley target for now? If so won't we just run those tests twice?

The makefile targets are for all components, but I am adding only a single job for the time being to bootstrap. Once I stabilize that, we can easily do the same for rest and remove the common targets & jobs.

> 
> Definitely think splitting it up is a good idea though.
> 
> We will need to add to testgrid though

That is a separate K8s repo with a separate check-in process.

File: tests/integration/tests.mk (hunk 4 above)
Comment (ozevren) [reply tests/integration/tests.mk]: race test change is just a move, not edit.

File: prow/integ-galley-local-presubmit-tests.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
Comment (utka) [inline prow/integ-galley-local-presubmit-tests.h]: Wrong file name .sh instead of .h

**Comment by nmittler** [comment]
@ozevren looks like prow/integ-galley-local-presubmit-tests.sh is broken ... it's trying to acquire a Boskos resource.  Not sure if it should be doing this for a local test?

**Comment by ozevren** [comment]
Yes, it was a last minute copy-paste bug. Fixing it here:
https://github.com/istio/istio/pull/13922


*From: *Nathan Mittler <notifications@github.com>
*Date: *Wed, May 8, 2019 at 12:46 PM
*To: *istio/istio
*Cc: *Ozben Evren, Mention

@ozevren <https://github.com/ozevren> looks like

**Pull request merged**